src/Cis194/Hw/Week1.hs

```diff
@@ -27,7 +33,20 @@ type Peg = String
 type Move = (Peg, Peg)
 
 hanoi :: Integer -> Peg -> Peg -> Peg -> [Move]
-hanoi _ _ _ _ = []
+hanoi 0 _ _ _ = []
+hanoi 1 s t _ = [(s, t)]
+hanoi n s t a = next n s a t ++
+     hanoi 1 s t a ++
+     next n a t s
```

Comment:
I'm still having to adjust to the order of operations in Haskell, since it wasn't immediately clear how this was working.

The fact that all multi-argument functions are actually implemented as a series of one-argument functions is just ingenious...even if it does throw me for a loop sometimes. Anyway, nice use of it here!